src/util/preemptive-rate-limit.ts

```diff
@@ -0,0 +1,52 @@
+import { createInstallationClient } from "utils/get-github-client-config";
+import { Octokit } from "@octokit/rest";
+import { numberFlag, NumberFlags } from "config/feature-flags";
+import { SQSMessageContext } from "~/src/sqs/sqs.types";
+import { SqsQueue } from "~/src/sqs/sqs";
+import Logger from "bunyan";
+
+// List of queues we want to apply the preemptive rate limiting on
+const TARGETTED_QUEUES = ["backfill"];
```

Comment:
Im pretty sure after my deployment errors look around today I'm going to want to add the deployments queue here too!!